src/Tempest/Router/src/RequestHeaders.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Router;
+
+use ArrayAccess;
+use ArrayIterator;
+use IteratorAggregate;
+use LogicException;
+use Traversable;
+
+final readonly class RequestHeaders implements ArrayAccess, IteratorAggregate
+{
+    /**
+     * @param array<string, string> $headers
+     */
+    public static function normalizeFromArray(array $headers): self
+    {
+        $normalized = array_combine(
+            array_map('strtolower', array_keys($headers)),
```

Comment:
I prefer to use `strtolower(...)`

Reply:
Done, I always forget about the syntax you mentioned, but very nice suggestion.